extensions/cpsection/language/model.py

```diff
@@ -81,49 +81,54 @@ def _initialize():
 
 
 def _write_i18n(lang_env, language_env):
-    path = os.path.join(os.environ.get('HOME'), '.i18n')
-    if not os.access(path, os.W_OK):
-        print _standard_msg
-        fd = open(path, 'w')
-        fd.write('LANG="%s"\n' % _default_lang)
-        fd.write('LANGUAGE="%s"\n' % _default_lang)
-        fd.close()
-    else:
-        fd = open(path, 'w')
-        fd.write('LANG="%s"\n' % lang_env)
-        fd.write('LANGUAGE="%s"\n' % language_env)
-        fd.close()
+    path = os.path.join(os.environ.get('HOME', ''), '.i18n')
+    # try avoid writing the file if the values didn't changed
+    lang_line = None
+    language_line = None
+    try:
+        with open(path) as fd:
+            for line in fd:
+                if line.startswith('LANG='):
+                    lang_line = line
+                elif line.startswith('LANGUAGE='):
+                    language_line = line
+    except:
+        pass
+
+    new_lang_line = 'LANG="%s"\n' % lang_env
+    new_language_line = 'LANGUAGE="%s"\n' % language_env
+
+    if lang_line == new_lang_line and language_line == new_language_line:
+        return
 
+    try:
+        with open(path, 'w') as fd:
+            fd.write(new_lang_line)
+            fd.write(new_language_line)
+            fd.flush()
+            # be sure all is flushed
+            os.fsync(fd)
+    except:
+        logging.exception('Error writing .i18n file')
```

Comment:
Line 112 logs an exception only.  There's no UI response, so the user will think the change will be effective, but it won't be.  Why not catch the exception in caller and revert the change?